quantum/send_string/send_string.h

```diff
@@ -106,6 +114,13 @@ void send_byte(uint8_t number);
  */
 void send_nibble(uint8_t number);
 
+/**
+ * \brief Wait the given number of milliseconds.
+ *
+ * \param interval The number of milliseconds to wait.
+ */
+void wait_delay(uint8_t interval);
```

Comment:
This function is badly named and should really live in `wait.c`.

Reply:
It's also not necessary as all the wait implementations were aligned to allow both variable and constant arguments

Reply:
Sorry, I don't understand. This doesn't change the behavior to sending more than one character. The existing code already sends several key events if the character is shifted.

The `send_string_with_delay` function already works with a delay time that is totally independent of the tap delay, so that e.g. `SEND_STRING_DELAY` can set an arbitrary delay between key presses. The problem is that the current code does not use a consistent delay between all the key presses because it doesn't use the input delay at all for ASCII chars. So e.g. if you do

```c
SEND_STRING_DELAY("tail -F file | grep \"hello\"", 20);
```

This will

1. wait 20ms for each of `tail -`
2. TAP_CODE_DELAY for shift down, `f`, shift up
3. 20ms for `file `
4. TAP_CODE_DELAY for shift down, `\`, shift up
5. 20ms for ` grep`
6. Etc, etc

>This function is badly named and should really live in wait.c

Ah, thanks for the pointer, I'm unfamiliar with the code base, so I'll take a look here.

Reply:
Actually it's just calling `register_code` directly, so it may not even be waiting the tap delay in between the mods and the ASCII character?